From 58a8a1f41bdbd500926b1f294b1b81709dda7466 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Wed, 13 Nov 2013 15:10:31 +0100
Subject: [PATCH] Update to Sisu 0.1.0

Sisu depends on Guice, but dependency scope changed from "compile" to
"provided" in Sisu 0.1.0.  As a Sisu user, XMvn needs to have runtime
dependency on Guice.
---
 xmvn-connector/pom.xml | 11 +++++++++++
 xmvn-core/pom.xml      | 11 +++++++++++
 xmvn-mojo/pom.xml      | 11 +++++++++++
 xmvn-parent/pom.xml    |  9 ++++++++-
 xmvn-tools/pom.xml     | 11 +++++++++++
 5 files changed, 52 insertions(+), 1 deletion(-)

diff --git a/xmvn-connector/pom.xml b/xmvn-connector/pom.xml
index 23a4cd8..e01219b 100644
--- a/xmvn-connector/pom.xml
+++ b/xmvn-connector/pom.xml
@@ -75,5 +75,16 @@
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
   </dependencies>
 </project>
diff --git a/xmvn-core/pom.xml b/xmvn-core/pom.xml
index 0b1da35..36ef433 100644
--- a/xmvn-core/pom.xml
+++ b/xmvn-core/pom.xml
@@ -50,6 +50,17 @@
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-utils</artifactId>
     </dependency>
diff --git a/xmvn-mojo/pom.xml b/xmvn-mojo/pom.xml
index cf7fbe2..d949bc1 100644
--- a/xmvn-mojo/pom.xml
+++ b/xmvn-mojo/pom.xml
@@ -72,6 +72,17 @@
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
       <groupId>org.eclipse.aether</groupId>
       <artifactId>aether-api</artifactId>
     </dependency>
diff --git a/xmvn-parent/pom.xml b/xmvn-parent/pom.xml
index 1f1a025..53ac52f 100644
--- a/xmvn-parent/pom.xml
+++ b/xmvn-parent/pom.xml
@@ -77,12 +77,13 @@
     <aetherVersion>0.9.0.M3</aetherVersion>
     <ivyVersion>2.3.0</ivyVersion>
     <jcommanderVersion>1.32</jcommanderVersion>
+    <guiceVersion>3.1.8</guiceVersion>
     <mavenInvokerVersion>2.1.1</mavenInvokerVersion>
     <mavenVersion>3.1.1</mavenVersion>
     <plexusUtilsVersion>3.0.15</plexusUtilsVersion>
     <plexusVersion>1.5.5</plexusVersion>
     <pluginToolsVersion>3.2</pluginToolsVersion>
-    <sisuVersion>0.0.0.M5</sisuVersion>
+    <sisuVersion>0.1.0</sisuVersion>
 
     <!-- Build dependencies -->
     <apivizVersion>1.3.2.GA</apivizVersion>
@@ -156,6 +157,12 @@
         <version>${sisuVersion}</version>
       </dependency>
       <dependency>
+        <groupId>org.sonatype.sisu</groupId>
+        <artifactId>sisu-guice</artifactId>
+        <version>${guiceVersion}</version>
+	<classifier>no_aop</classifier>
+      </dependency>
+      <dependency>
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-utils</artifactId>
         <version>${plexusUtilsVersion}</version>
diff --git a/xmvn-tools/pom.xml b/xmvn-tools/pom.xml
index c37b109..0aebd8d 100644
--- a/xmvn-tools/pom.xml
+++ b/xmvn-tools/pom.xml
@@ -51,6 +51,17 @@
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
   </dependencies>
   <build>
     <plugins>
-- 
1.8.3.1

